src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -11,49 +11,154 @@ import "@material/mwc-button";
 import { HomeAssistant } from "../../../../types";
 import { LovelaceCardConfig } from "../../../../data/lovelace";
 import { getCardElementTag } from "../../common/get-card-element-tag";
-import { CardPickTarget } from "../types";
+import { CardPickTarget, CardOption } from "../types";
 import { fireEvent } from "../../../../common/dom/fire_event";
-
-const cards = [
-  { name: "Alarm panel", type: "alarm-panel" },
-  { name: "Conditional", type: "conditional" },
-  { name: "Entities", type: "entities" },
-  { name: "Entity Button", type: "entity-button" },
-  { name: "Entity Filter", type: "entity-filter" },
-  { name: "Gauge", type: "gauge" },
-  { name: "Glance", type: "glance" },
-  { name: "History Graph", type: "history-graph" },
-  { name: "Horizontal Stack", type: "horizontal-stack" },
-  { name: "iFrame", type: "iframe" },
-  { name: "Light", type: "light" },
-  { name: "Map", type: "map" },
-  { name: "Markdown", type: "markdown" },
-  { name: "Media Control", type: "media-control" },
-  { name: "Picture", type: "picture" },
-  { name: "Picture Elements", type: "picture-elements" },
-  { name: "Picture Entity", type: "picture-entity" },
-  { name: "Picture Glance", type: "picture-glance" },
-  { name: "Plant Status", type: "plant-status" },
-  { name: "Sensor", type: "sensor" },
-  { name: "Shopping List", type: "shopping-list" },
-  { name: "Thermostat", type: "thermostat" },
-  { name: "Vertical Stack", type: "vertical-stack" },
-  { name: "Weather Forecast", type: "weather-forecast" },
-];
-
 @customElement("hui-card-picker")
 export class HuiCardPicker extends LitElement {
   public hass?: HomeAssistant;
 
   public cardPicked?: (cardConf: LovelaceCardConfig) => void;
 
   protected render(): TemplateResult | void {
+    const cards: CardOption[] = [
```

Comment:
Why was this moved to `render`? This is not needed to be recalculated every render.
You can move it to updated and update it only if the language is changed.

Reply:
Better yet, we don't need the name here, we can just use this in the render:
```js
this.hass!.localize(`ui.panel.lovelace.editor.card.${card.type}.name`)
```

Reply:
Decided to remove the object entirely in favor for a string array. and use your above suggestion